packages/medusa/src/models/line-item-adjustment.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Entity,
+  BeforeInsert,
+  Index,
+  Column,
+  ManyToOne,
+  JoinColumn,
+  PrimaryColumn,
+} from "typeorm"
+import { ulid } from "ulid"
+import { DbAwareColumn } from "../utils/db-aware-column"
+import { LineItem } from "./line-item"
+
+@Entity()
+export class LineItemAdjustment {
+  @PrimaryColumn()
+  id: string
+
+  @Index()
+  @Column()
+  item_id: string
+
+  @ManyToOne(() => LineItem, (li) => li.adjustments)
+  @JoinColumn({ name: "item_id" })
+  item: LineItem
+
+  @Column()
+  description: string
+
+  @Column({ nullable: true })
+  resource_id: string
```

Comment:
Correct. What suggested makes a lot sense! One thing I want to double check with you is whether the `resource_id` is always supposed to refer to a discount id. If that's the case, then making it as a foreign key sounds good. I remember @olivermrbl had suggested the `resource_id` and not something like `discount_id` so I assumed there could be use cases where line items are "adjusted" from something other than just discounts. In case line item adjustments are only supposed to come from discounts, wouldn't it make more sense to rename the `resource_id` to something like `discount_id`?

Another use case we talked about in [#110](https://github.com/medusajs/medusa-engineering/issues/110) is the creation of "custom" line item adjustments that Kasper highlighted. However, I believe that making the foreign key as nullable would suffice to cover this use case. 

Let me know what you think @srindom @olivermrbl 😄

Reply:
To the first point: I could imagine it would make sense to also have `gift_card_id` at some point so let's definitely change it to `discount_id`.

And yes: the FK should be nullable so that a custom adjustment is possible :)